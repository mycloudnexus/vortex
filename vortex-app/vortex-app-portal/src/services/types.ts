export interface ICompany {
  name: string
  id: string
  display_name: string
  connection: Connection
  metadata: MetaData
  branding: Branding
}

interface Connection {
  name: string
  strategy: string
  display_name: string
  options: {
    additionalProp1: Object
    additionalProp2: Object
    additionalProp3: Object
  }
  id: string
  enabled_clients: string[]
  provisioning_ticket_url: string
  metadata: {
    additionalProp1: string
    additionalProp2: string
    additionalProp3: string
  }
  realms: string
}

interface MetaData {
  status: string
  strategy: string
  connectionId: string
}

interface Branding {
  logo_url: string
  colors: Colors
}

interface Colors {
  primary: string
  page_background: string
}

export interface CreateOrganizationRequestBody {
  name: string
  display_name: string
}

export interface CreateOrganizationResponse {
  code: number
  message: string
  data: ICompany
}

export interface IOrganization {
  data: ICompany[]
  page: number
  size: number
  total: number
}

export interface RequestResponse<Data> {
  code: number
  message: string
  data: Data
}

export interface UpdateOrganizationRequestBody {
  request_body: { display_name?: string; status?: string }
  id: string
}

interface Samlp {
  signingCert: string
  signSAMLRequest: boolean
  signatureAlgorithm: string
  digestAlgorithm: string
  fieldsMap: Object
  signInEndpoint: string
  signOutEndpoint: string
  debug: boolean
}

export interface AddConnectionRequestBody {
  strategy: string
  saml: Samlp
}

export interface AddConnectionResponse {
  name: string
  strategy: string
  options: Options
  id: string
  enabled_clients: string[]
  provisioning_ticket_url: string
  realms: string[]
}

export interface Options {
  signInEndpoint: string
  signingCert: string
  debug: boolean
  signOutEndpoint: string
  signSAMLRequest: boolean
  digestAlgorithm: string
  signatureAlgorithm: string
  fieldsMap: FieldsMap
  expires: Date
  subject: Subject
  thumbprints: string[]
  cert: string
}

export interface FieldsMap {}

export interface Subject {
  commonName: string
}
