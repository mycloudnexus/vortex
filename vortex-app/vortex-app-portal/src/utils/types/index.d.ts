declare module '*.svg' {
  import * as React from 'react'

  export const ReactComponent: React.FunctionComponent<React.SVGProps<SVGSVGElement> & { title?: string }>

  const src: string
  export default src
}

declare module '*.module.css'
declare module '*.module.scss'
declare module '*.png'
declare interface Window {
  portalConfig: Record<string, unknown>
  portalLoggedInUser: Record<string, unknown>
  portalAccessRoles: Record<string, unknown>
  portalToken: string
}
