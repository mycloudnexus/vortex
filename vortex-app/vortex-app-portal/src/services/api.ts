export const USER_INFO = '/api/user'
export const AUTH_TOKEN = '/api/auth/token'
export const USER_ROLE = '/v2/admin/roles'

//Customer Company Endpoints
export const ORGANIZATIONS = '/api/mgmt/organizations'
