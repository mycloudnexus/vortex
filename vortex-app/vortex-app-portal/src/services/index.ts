import request from '@/utils/helpers/request'
import { USER_INFO, AUTH_TOKEN, USER_ROLE } from './api'

export const getUserDetail = (name: string) => {
  return request(`${USER_INFO}/${name}`, {})
}

export const getUserAuthToken = () => {
  return request(AUTH_TOKEN)
}

export const getUserRole = () => {
  return request(USER_ROLE)
}
