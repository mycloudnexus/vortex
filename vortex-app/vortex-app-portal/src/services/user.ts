import request from '@/utils/helpers/request'
import { DOWNSTREAM_USER_INFO, DOWNSTREAM_DOWNSTREAM_USER_INFO, DOWNSTREAM_USER_ROLE, VORTEX_USER_ROLE } from './api'

export const getUserDetail = (name: string) => {
  return request(`${DOWNSTREAM_USER_INFO}/${name}`, {})
}

export const getUserAuthToken = () => {
  return request(DOWNSTREAM_DOWNSTREAM_USER_INFO)
}

export const getUserRole = () => {
  return request(DOWNSTREAM_USER_ROLE)
}

export const getVortexUser = () => {
  return request(VORTEX_USER_ROLE)
}
