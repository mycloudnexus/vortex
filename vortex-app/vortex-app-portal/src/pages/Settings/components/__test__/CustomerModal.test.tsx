import { render, fireEvent, waitFor } from '@testing-library/react'
import { Form } from 'antd'
import CustomerCompanyModal from '../CustomerModal'
import { ICompany } from '@/services/types'

jest.mock('@/assets/icon/info.svg', () => ({
  ReactComponent: () => <svg />
}))

global.matchMedia =
  global.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: jest.fn(),
      removeListener: jest.fn()
    }
  }

const mockCompanies: ICompany[] = [
  {
    branding: {
      colors: { page_background: '', primary: '' },
      logo_url: ''
    },
    display_name: 'Company A',
    id: '123',
    metadata: {
      loginType: '',
      status: 'active',
      type: ''
    },
    name: 'test'
  },
  {
    branding: {
      colors: { page_background: '', primary: '' },
      logo_url: ''
    },
    display_name: 'Company B',
    id: '2',
    metadata: {
      loginType: '',
      status: '',
      type: ''
    },
    name: 'test'
  }
]

interface FormWrapperProps {
  handleOk: () => void
  handleCancel: () => void
  type?: 'add' | 'update'
}

const FormWrapper = ({ handleOk, handleCancel, type = 'add' }: FormWrapperProps) => {
  const [form] = Form.useForm()

  return (
    <CustomerCompanyModal
      title='Modify customer'
      name='modify_customer_company'
      companies={mockCompanies}
      form={form}
      handleCancel={handleCancel}
      handleOk={handleOk}
      initialValues={{ title: '', shortName: '' }}
      isModalOpen={true}
      type={type}
    />
  )
}

describe('CustomerCompanyModal', () => {
  it('renders the modal correctly with title and inputs', () => {
    const { getByText, getByLabelText, baseElement } = render(
      <FormWrapper handleOk={jest.fn()} handleCancel={jest.fn()} />
    )

    expect(getByText('Modify customer')).toBeInTheDocument()
    expect(getByLabelText(/Customer name/)).toBeInTheDocument()
    expect(getByLabelText(/Customer URL short name/)).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })

  it('should call handleSubmit when the OK button is clicked', () => {
    const handleSubmitMock = jest.fn()
    const { getByText } = render(<FormWrapper handleCancel={jest.fn()} handleOk={handleSubmitMock} />)
    fireEvent.click(getByText('OK'))
    expect(handleSubmitMock).toHaveBeenCalledTimes(1)
  })
  it('should call handleCancel when the cancel button is clicked', () => {
    const handleCancelMock = jest.fn()
    const { getByRole } = render(<FormWrapper handleCancel={handleCancelMock} handleOk={jest.fn()} />)
    fireEvent.click(getByRole('button', { name: /cancel/i }))
    expect(handleCancelMock).toHaveBeenCalledTimes(1)
  })

  it('should disable the short name input when type is "update"', () => {
    const { getByLabelText } = render(<FormWrapper handleOk={jest.fn()} handleCancel={jest.fn()} type='update' />)
    expect(getByLabelText(/Customer URL short name/)).toBeDisabled()
  })

  it('should show an error message if the customer name is unique', async () => {
    const { getByLabelText, getByText, getByTestId } = render(
      <FormWrapper handleOk={jest.fn()} handleCancel={jest.fn()} />
    )

    fireEvent.change(getByTestId('customer-name'), { target: { value: 'Company B' } })
    expect(getByLabelText(/Customer name/)).toBeInTheDocument()
    fireEvent.click(getByText('OK'))

    await waitFor(() => {
      expect(getByText(/Customer name cannot be duplicated/i)).toBeInTheDocument()
    })
  })

  it('should not show an error message if the customer name is unique', async () => {
    const { getByLabelText, getByText, getByTestId, queryByText } = render(
      <FormWrapper handleOk={jest.fn()} handleCancel={jest.fn()} />
    )

    fireEvent.change(getByTestId('customer-name'), { target: { value: 'Company C' } })
    expect(getByLabelText(/Customer name/)).toBeInTheDocument()
    fireEvent.click(getByText('OK'))

    await waitFor(() => {
      expect(queryByText(/Customer name cannot be duplicated/i)).toBeNull()
    })
  })
})
