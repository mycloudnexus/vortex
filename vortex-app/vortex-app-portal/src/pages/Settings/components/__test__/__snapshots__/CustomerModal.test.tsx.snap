// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCompanyModal renders the modal correctly with title and inputs 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-ni1kz0"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-ni1kz0 sc-blHHSb jFpWO ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Modify Customer company
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <form
                  class="ant-form ant-form-vertical css-dev-only-do-not-override-ni1kz0"
                  id="modify_customer_company"
                  style="padding: 20px;"
                >
                  <div
                    class="ant-form-item css-dev-only-do-not-override-ni1kz0"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-ni1kz0"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-ni1kz0"
                      >
                        <label
                          class=""
                          for="modify_customer_company_display_name"
                          title=""
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            Customer company name
                            <span
                              style="color: red;"
                            >
                              *
                            </span>
                          </div>
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-ni1kz0"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input css-dev-only-do-not-override-ni1kz0 ant-input-outlined sc-gtLWhw jtrSU"
                              data-testid="customer-name"
                              id="modify_customer_company_display_name"
                              maxlength="255"
                              placeholder="Please enter"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-form-item css-dev-only-do-not-override-ni1kz0"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-ni1kz0"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-ni1kz0"
                      >
                        <label
                          class=""
                          for="modify_customer_company_name"
                          title=""
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            Customer company URL short name
                            <span
                              data-testid="mocked-svg"
                              style="display: flex; justify-content: center;"
                            >
                              <svg />
                            </span>
                            <span
                              style="color: red;"
                            >
                              *
                            </span>
                          </div>
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-ni1kz0"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-describedby="modify_customer_company_name_extra"
                              class="ant-input css-dev-only-do-not-override-ni1kz0 ant-input-outlined sc-gtLWhw jtrSU"
                              id="modify_customer_company_name"
                              maxlength="20"
                              placeholder="Please enter"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="ant-form-item-extra"
                          id="modify_customer_company_name_extra"
                        >
                          Only lower case letters, numbers, - , allowed. No more than 20 characters in total
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="ant-modal-footer"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
                  type="button"
                >
                  <span>
                    Cancel
                  </span>
                </button>
                <button
                  class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
                  type="button"
                >
                  <span>
                    OK
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
